decompiler/pipeline/controlflowanalysis/loop_utility_methods.py

```diff
@@ -210,3 +228,89 @@ def _requirement_without_reinitialization(ast: AbstractSyntaxTree, node: Abstrac
                 return True
             elif variable in assignment.requirements:
                 return True
+
+
+def _get_continue_nodes_with_equalizable_definition(
+    loop_node: WhileLoopNode, continuation: AstInstruction, variable_init: AstInstruction
+) -> List[CodeNode]:
+    """
+    Finds code nodes of a while loop containing continue statements and a definition of the continuation instruction, which can be easily equalized.
+
+    :param loop_node: While-loop to search in
+    :param continuation: Instruction defining the for-loops modification
+    :param variable_init: Instruction defining the for-loops declaration
+    :return: List of continue code nodes that has equalizable definitions, None if at least one continue node does not match the requirements
+    """
+    equalizable_nodes = []
+    for code_node in (
+        node for node in loop_node.body.get_descendant_code_nodes_interrupting_ancestor_loop() if node.does_end_with_continue
+    ):
+        if not _is_expression_simple_binary_operation(continuation.instruction.value):
+            return None
+        if (last_definition_index := _get_last_definition_index_of(code_node, variable_init.instruction.destination)) == -1:
+            return None
+        if not (
+            isinstance(code_node.instructions[last_definition_index].value, Constant)
+            or _is_expression_simple_binary_operation(code_node.instructions[last_definition_index].value)
+            and _get_variable_in_binary_operation(continuation.instruction.value)
+            == _get_variable_in_binary_operation(code_node.instructions[last_definition_index].value)
```

Comment:
Again, you check only if they contain the same variable. However, this works if your checks before are correct. But, instead of computing them multiple times, it could be an option to compute them once and save them.

Reply:
Fixed in 29ec9fa. The checks before were corrected, but I also changed the `_get_variable_in_binary_operation` function to iterate over the operands and don't use `subexpressions()`.